docs/how-tos/2_save_variables_and_secrets.md

```diff
@@ -9,11 +9,28 @@ When writing scripts, you may want to reuse variables, or safely pass secrets to
 scripts. You can do that with **Variables**. Windmill has user-defined variables
 and reserved variables.
 
+:::caution
+
+Secrets are encrypted when stored on Windmill. From a usage standpoint, secrets
+are kept safe in three different ways:
+
+- Secrets can only be access by users with the right permissions, as defined by
+  their path. In addition, secrets can be explicitly shared with users or
+  groups. A secret in `u/alice/secret` will only be accessible by `alice`,
+  unless explicitly shared.
+- Secrets cannot be viewed outside of scripts. Note that a user could still
+  `print` a secret they have access to from a script.
+- However, each script run generates an audit log, and scripts are versioned.
+  That means than you can audit, who accesses secrets. For example, a user
+  writing a script to print all secrets would be traceable thanks to audit logs.
```

Comment:
From within a script, accessing a secret generate a dedicated audit log

(it was not the case before as all secrets were just passed on as env variable)